certbot/compat.py

```diff
@@ -199,3 +199,19 @@ def get_default_folder(folder_type):
         return LINUX_DEFAULT_FOLDERS[folder_type]
     # Windows specific
     return WINDOWS_DEFAULT_FOLDERS[folder_type]
+
+
+def normalize_path(path):
```

Comment:
I'm a little on the fence about adding this function here. Do we expect to have other paths that are incompatible with Windows that need fixing up and do we expect that we'll want to make the same decision about how to modify them everywhere?

Unless we're confident about both, I think I'd personally prefer to either put this code in the same file that's modifying the account path or rename this function to specifically mention accounts paths. I think I prefer the former for now as we can always move code into `compat` if we have multiple uses for it later.

What do you think?

Reply:
For this code specifically, yes you are right, the method could be replaced by an inline logic, and I am not expecting to correct paths somewhere else.

But in fact my motivation to put the logic as a separate function follows a design convention that I tried to respect so far: for better maintainancy, I put every platform specific code in the compat module, to let only platform agnostic code everywhere else (except tests, to avoid to clutter the module with non-functional logic).

In fact with the windows permission PR, this compat module will become a package, to improve modularization of platform specific code. And I am also thinking to move the lockfile module in this package when the relevant PR will be integrated and compat package module is available.

Even if it adds some lines in particular implementations, I think it is really important to keep this platform specific code in one place in the long term.

Reply:
I think however that the function name can be changed to something more specific about replacing platform specific unsupported characters in paths.

Reply:
That sounds good to me. I was initially not worried about moving code like this out of `compat` because we're already doing that in the locking PR, but if you're planning to move that code back in later, that makes sense to me and I agree a more explicit name here is good enough.